### Logistic Regression Analysis

In [2]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
import numpy as np

In [3]:
df = pd.read_csv("LOWBWT.csv",)
df.head()

ID  AGE  LWT  RACE  SMOKE  PTL  HT  UI  FTV   BWT
0  85   19  182     2      0    0   0   1    0  2523
1  86   33  155     3      0    0   0   0    3  2551
2  87   20  105     1      1    0   0   0    1  2557
3  88   21  108     1      1    0   0   1    2  2594
4  89   18  107     1      1    0   0   1    0  2600

#### Variables

* lwt : weight in last menstrual cycle in pounds

* ftv : visit to the physician

* ptl : previous pre termed labour

* smoke : smoking during

* ht : hypertension during preganancy

* race

    * 1=white
    * 2=black
    * 3=others

* ui : uterine irritability

* bwt : birth weight in grams

* categorize birth weight to :

    * low bwt <2500gmhigh 
    * bwt >=2500gm

In [4]:
df1 = df

In [5]:
for i in range(len(df1)):
    if df1.loc[i,'BWT'] >= 2500:
        df1.loc[i,'BWT1'] = 1
    else:
        df1.loc[i,'BWT1'] = 0

In [6]:
df1['BWT1'] = df1['BWT1'].astype('int')

In [7]:
df1['BWT1'] = df1['BWT1'].astype('str')

In [8]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 189 entries, 0 to 188
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   ID      189 non-null    int64 
 1   AGE     189 non-null    int64 
 2   LWT     189 non-null    int64 
 3   RACE    189 non-null    int64 
 4   SMOKE   189 non-null    int64 
 5   PTL     189 non-null    int64 
 6   HT      189 non-null    int64 
 7   UI      189 non-null    int64 
 8   FTV     189 non-null    int64 
 9   BWT     189 non-null    int64 
 10  BWT1    189 non-null    object
dtypes: int64(10), object(1)
memory usage: 16.4+ KB


In [9]:
df1.RACE = df1.RACE.astype('str')
df1.HT = df1.HT.astype('str')
df1.SMOKE = df1.SMOKE.astype('str')
df1.UI = df1.UI.astype('str')
df1.BWT = df1.BWT.astype('str')

In [10]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 189 entries, 0 to 188
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   ID      189 non-null    int64 
 1   AGE     189 non-null    int64 
 2   LWT     189 non-null    int64 
 3   RACE    189 non-null    object
 4   SMOKE   189 non-null    object
 5   PTL     189 non-null    int64 
 6   HT      189 non-null    object
 7   UI      189 non-null    object
 8   FTV     189 non-null    int64 
 9   BWT     189 non-null    object
 10  BWT1    189 non-null    object
dtypes: int64(5), object(6)
memory usage: 16.4+ KB


In [11]:
import statsmodels.api as sm
df1.drop(["BWT"],axis=1,inplace=True)
df1

ID  AGE  LWT RACE SMOKE  PTL HT UI  FTV BWT1
0    85   19  182    2     0    0  0  1    0    1
1    86   33  155    3     0    0  0  0    3    1
2    87   20  105    1     1    0  0  0    1    1
3    88   21  108    1     1    0  0  1    2    1
4    89   18  107    1     1    0  0  1    0    1
..   ..  ...  ...  ...   ...  ... .. ..  ...  ...
184  79   28   95    1     1    0  0  0    2    0
185  81   14  100    3     0    0  0  0    2    0
186  82   23   94    3     1    0  0  0    0    0
187  83   17  142    2     0    0  1  0    0    0
188  84   21  130    1     1    0  1  0    3    0

[189 rows x 10 columns]

In [12]:
df1.BWT1 = df1.BWT1.astype('int')

#### BWT vs AGE

In [13]:
model1 = smf.logit(formula="BWT1 ~ AGE", data=df1
                       ).fit()

Optimization terminated successfully.
         Current function value: 0.613524
         Iterations 5


In [14]:
model1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.01176
Time:                        09:39:31   Log-Likelihood:                -115.96
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.09665
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3846      0.732     -0.525      0.599      -1.820       1.050
AGE            0.0512      0.032      1.623      0.105      -0.011       0.113
==============================================================================
"""

In [15]:
model1.summary2()

<class 'statsmodels.iolib.summary2.Summary'>
"""
                        Results: Logit
===============================================================
Model:              Logit            Method:           MLE     
Dependent Variable: BWT1             Pseudo R-squared: 0.012   
Date:               2024-01-27 09:39 AIC:              235.9120
No. Observations:   189              BIC:              242.3955
Df Model:           1                Log-Likelihood:   -115.96 
Df Residuals:       187              LL-Null:          -117.34 
Converged:          1.0000           LLR p-value:      0.096646
No. Iterations:     5.0000           Scale:            1.0000  
----------------------------------------------------------------
              Coef.   Std.Err.     z     P>|z|    [0.025  0.975]
----------------------------------------------------------------
Intercept    -0.3846    0.7321  -0.5253  0.5994  -1.8195  1.0504
AGE           0.0512    0.0315   1.6232  0.1045  -0.0106  0.1129
===============================================================

"""

In [16]:
# crude odds ratio confidence interval for age
c1 = model1.conf_int(alpha=0.05)
np.exp(c1)

0         1
Intercept  0.162103  2.858671
AGE        0.989443  1.119541

In [17]:
# crude odds ratio for age
np.exp(0.0512)

1.0525333789097215

For an increase of 1 year in age of mother there is 5% more likelihood of Low Birth Weight in the mother.

#### BWT vs LWT

In [18]:
model2 = smf.logit(formula="BWT1 ~ LWT", data=df1
                       ).fit()
print(model2.summary())
print(model2.summary2())

# crude odds ratio confidence interval for LWT
c2 = model2.conf_int(alpha=0.05)
np.exp(c2)

# crude odds ratio for LWT
print(np.round(np.exp(0.0141),2))

Optimization terminated successfully.
         Current function value: 0.605002
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.02549
Time:                        09:39:31   Log-Likelihood:                -114.35
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.01446
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.9983      0.785     -1.271      0.204      -2.537       0.541
LWT            0.0141      0.

for increase of 1 pound of weight in last menstrual cycle there is 1% more likelihood of low birth weight in the mother

In [19]:
model3 = smf.logit(formula="BWT1 ~ RACE", data=df1
                       ).fit()
print(model3.summary())
print(model3.summary2())

# crude odds ratio confidence interval for LWT
c3 = model3.conf_int(alpha=0.05)
np.exp(c3)

# crude odds ratio for LWT
print(np.round(np.exp(-0.8448),2))
print(np.round(np.exp(-0.6362),2))

Optimization terminated successfully.
         Current function value: 0.607570
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      186
Method:                           MLE   Df Model:                            2
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.02135
Time:                        09:39:31   Log-Likelihood:                -114.83
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.08166
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.1550      0.239      4.830      0.000       0.686       1.624
RACE[T.2]     -0.8448      0.

by default Python takes the least code category as reference category so we have values of race2 and race3 relative to race1

race 1 -> OR =1 (reference) [no confidence interval]

race 2 -> OR interval

race 3 -> OR interval

A mother belonging to race 2 has 53% less likelihood of having low birth weight in comparison to one belonging to race 1. Similarly, a mother belonging to race 3 has 43% less likelihood of having low birth weight in comparison to one belonging to race 1

#### BWT vs SMOKE

In [20]:
model4 = smf.logit(formula="BWT1 ~ SMOKE", data=df1
                       ).fit()
print(model4.summary())
print(model4.summary2())

# crude odds ratio confidence interval for LWT
c4 = model4.conf_int(alpha=0.05)
np.exp(c4)

# crude odds ratio for LWT
print(np.round(np.exp(-0.7041),2))

Optimization terminated successfully.
         Current function value: 0.607949
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.02074
Time:                        09:39:31   Log-Likelihood:                -114.90
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.02737
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.0871      0.215      5.062      0.000       0.666       1.508
SMOKE[T.1]    -0.7041      0.

there is 50% less likelihood of low birth weight in mothers who smoked during menstrual cycle compared to mothers who did not smoke.

### BWT vs PTL

In [21]:
model5 = smf.logit(formula="BWT1 ~ PTL", data=df1).fit()
print(model5.summary())
print(model5.summary2())

# crude odds ratio confidence interval for LWT
c5 = model5.conf_int(alpha=0.05)
np.exp(c5)

# crude odds ratio for LWT
print(np.round(np.exp(-0.8081),2))

Optimization terminated successfully.
         Current function value: 0.602891
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.02889
Time:                        09:39:31   Log-Likelihood:                -113.95
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                  0.009222
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.9642      0.175      5.511      0.000       0.621       1.307
PTL           -0.8018      0.

for mothers with PTL (greater than 0 number of days) there is 45% less likelihood of low birth weight in mothers compared to mothers with no pretermed labour.

### BWT vs HT

In [22]:
model6 = smf.logit(formula="BWT1 ~ HT", data=df1).fit()
print(model6.summary())
print(model6.summary2())

# crude odds ratio confidence interval for LWT
c6 = model6.conf_int(alpha=0.05)
np.exp(c6)

# crude odds ratio for LWT
print(np.round(np.exp(-1.2135),2))

Optimization terminated successfully.
         Current function value: 0.610185
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.01714
Time:                        09:39:32   Log-Likelihood:                -115.32
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.04491
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8771      0.165      5.315      0.000       0.554       1.200
HT[T.1]       -1.2135      0.

mother with hypertension during pregnancy have 70% less likelihood of low birth weight compared to mothers without hypertension

### BWT vs UI

In [23]:
model7 = smf.logit(formula="BWT1 ~ UI", data=df1).fit()
print(model7.summary())
print(model7.summary2())

# crude odds ratio confidence interval for LWT
c7 = model7.conf_int(alpha=0.05)
np.exp(c7)

# crude odds ratio for LWT
print(np.round(np.exp(-0.9469),2))

Optimization terminated successfully.
         Current function value: 0.607397
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                 0.02163
Time:                        09:39:32   Log-Likelihood:                -114.80
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                   0.02426
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.9469      0.176      5.392      0.000       0.603       1.291
UI[T.1]       -0.9469      0.

for mothers who have had uterine irritability have 61 % less likelihood of low birth weight compared to those who did not have

### BWT vs FTV

In [24]:
model8 = smf.logit(formula="BWT1 ~ FTV", data=df1).fit()
print(model8.summary())
print(model8.summary2())

# crude odds ratio confidence interval for LWT
c8 = model8.conf_int(alpha=0.05)
np.exp(c8)

# crude odds ratio for LWT
print(np.round(np.exp(0.1351),2))

Optimization terminated successfully.
         Current function value: 0.618780
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      187
Method:                           MLE   Df Model:                            1
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                0.003295
Time:                        09:39:32   Log-Likelihood:                -116.95
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                    0.3792
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.6868      0.195      3.525      0.000       0.305       1.069
FTV            0.1351      0.

mothers who visited physicians for have 14% more likelihood of low birth weight relative to mothers who didn't visit them.

### Multiple Logistic Regression

In [25]:
model9 = smf.logit(formula="BWT1 ~ AGE+FTV+RACE+SMOKE+PTL+HT+UI", data=df1).fit()
print(model9.summary())
print(model9.summary2())

# crude odds ratio confidence interval for LWT
c9 = model9.conf_int(alpha=0.05)
np.exp(c9)

# crude odds ratio for LWT
print(np.round(np.exp(model9.params),2))

Optimization terminated successfully.
         Current function value: 0.547102
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:                   BWT1   No. Observations:                  189
Model:                          Logit   Df Residuals:                      180
Method:                           MLE   Df Model:                            8
Date:                Sat, 27 Jan 2024   Pseudo R-squ.:                  0.1188
Time:                        09:39:32   Log-Likelihood:                -103.40
converged:                       True   LL-Null:                       -117.34
Covariance Type:            nonrobust   LLR p-value:                 0.0005001
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.1833      0.919      1.288      0.198      -0.617       2.984
RACE[T.2]     -1.0092      0.

### Report

#### Descriptives

In [30]:
s1 = df1.loc[df1["BWT1"]==0,]
s2 = df1.loc[df1["BWT1"]==1,]

In [35]:
s1.describe()

ID        AGE         LWT        PTL        FTV  BWT1
count  59.000000  59.000000   59.000000  59.000000  59.000000  59.0
mean   44.932203  22.305085  122.135593   0.338983   0.694915   0.0
std    22.382155   4.511496   26.559275   0.544888   1.038139   0.0
min     4.000000  14.000000   80.000000   0.000000   0.000000   0.0
25%    26.500000  19.500000  104.000000   0.000000   0.000000   0.0
50%    44.000000  22.000000  120.000000   0.000000   0.000000   0.0
75%    62.500000  25.000000  130.000000   1.000000   1.000000   0.0
max    84.000000  34.000000  200.000000   2.000000   4.000000   0.0

In [37]:
print(s1.RACE.value_counts())

RACE
3    25
1    23
2    11
Name: count, dtype: int64


In [33]:
s2.describe()

ID         AGE         LWT         PTL         FTV   BWT1
count  130.000000  130.000000  130.000000  130.000000  130.000000  130.0
mean   155.638462   23.661538  133.300000    0.130769    0.838462    1.0
std     41.944778    5.584522   31.724016    0.455602    1.069694    0.0
min     85.000000   14.000000   85.000000    0.000000    0.000000    1.0
25%    119.250000   19.000000  113.000000    0.000000    0.000000    1.0
50%    154.500000   23.000000  123.500000    0.000000    1.000000    1.0
75%    191.750000   28.000000  147.000000    0.000000    1.000000    1.0
max    226.000000   45.000000  250.000000    3.000000    6.000000    1.0

In [38]:
print(s2.RACE.value_counts())

RACE
1    73
3    42
2    15
Name: count, dtype: int64


| Variables | Low Birth Weight           | High Birth Weight          |
|-----------|----------------------------|----------------------------|
| Age       | 22.31 +/- 4.5              | 23.66 +/- 5.58             |
| LWT       | 122.1 +/- 26.56            | 133.3 +/- 31.72            |
| RACE1     | 23 (24%)                   | 73 (76%)                   |
| RACE2     | 11 (42.31%)                | 15 (57.69%)                 |
| RACE3     | 25 (37.31%)                | 42 (62.69%)                |
| SMOKE0    | 29 (25.22%)                | 86 (74.78%)                |
| SMOKE1    | 30 (40.54%)                | 44 (59.46%)                |
| PTL       | Median = 0 (IQR = 1-0 =1)  | Median = 0 (IQR = 0-0 = 0) |
| FTV       | Median = 1 (IQR = 2-1 = 1) | Median = 2 (IQR = 2-1 = 1) |
| HI0       | 125 (70.62%)               | 52 (29.38%)                |
| HI1       | 7 (58.33%)                 | 5 (41.67%)                 |
| UI0       | 45 (27.95 %)               | 116 (72.05 %)              |
| UI1       | 14 (0.5 %)                   | 14 (0.5 %)               |

| Variables | Crude Odds Ratio | 95% confidence interval  | Adjusted Odds Ratio | 95% confidence interval  |
|---------------|---------------|---------------|---------------|---------------|
| Age       | 1.052481         | (0.9910538 , 1.121963)   | 1.0421249           | (0.97135309 , 1.1212775) |
| FTV       | 1.142022         | (0.8448423 , 1.582761)   | 0.9790425           | (0.70894787, 1.3680476)  |
| RACE1     | 1                | No confidence interval   | 1                   | No Confidence Interval   |
| RACE2     | 0.4296433        | (0.1731692 , 1.080437)   | 0.3645071           | (0.13478907 , 0.9815848) |
| RACE3     | 0.5292999        | (0.2661081 , 1.045432)   | 0.3656779           | (0.15479810 , 0.8307541) |
| SMOKE0    | 1                | No Confidence interval   | 1                   | No confidence interval   |
| SMOKE1    | 0.4945535        | (0.2631176 , 0.9243235)  | 0.3806482           | (0.17328552 , 0.8095628) |
| HT0       | 1                | No Confidence interval   | 1                   | No confidence Interval   |
| HT1       | 0.2971554        | (0.08453716 , 0.9724023) | 0.2552515           | (0.06954948 , 0.8721817) |
| UI0       | 1                | No Confidence interval   | 1                   | No confidence interval   |
| UI1       | 0.3879418        | (0.1700315 , 0.8824552)  | 0.4478090           | (0.18215164 , 1.1109683) |
| PTL       | 0.4485209        | (0.2334474, 0.8210889)   | 0.5324392           | (0.26610261 , 1.0248160) |


## Interpretation

### Crude Odds Ratio

there is 50% less likelihood of low birth weight in child who smoked during menstrual cycle compared to mothers who did not smoke.

For an increase of 1 year in age of mother there is 5% more likelihood of Low Birth Weight in the child.

### Adjusted Odds Ratio

there is 50% less likelihood of low birth weight in mothers who smoked during menstrual cycle compared to mothers who did not smoke, adjusted for all other variables included in the model

For an increase of 1 year in age of mother there is 5% more likelihood of causing Low Birth Weight in the mother, adjusted for all other variables included in the model.